R/fill.R

```diff
@@ -19,8 +19,94 @@ NULL
 #'   or "updown" (first up and then down).
 #' @export
 #' @examples
-#' df <- data.frame(Month = 1:12, Year = c(2000, rep(NA, 11)))
-#' df %>% fill(Year)
+#' # Here we have some spreadsheet data where the analyst has only updated year
+#' # when the year has changed, leading to a consistent placement of year at the top
+#' # followed by trailing NAs up to the next year
+#'
+#' sales <- tibble(quarter = rep(c("Q1", "Q2", "Q3", "Q4"), 4),
+#'                 year = c(2000, rep(NA, 3),
+#'                          2001, rep(NA, 3),
+#'                          2002, rep(NA, 3),
+#'                          2004, rep(NA, 3)
+#'                 ),
+#'                 sales = c(66013, 69182, 53175, 21001, 46036, 58842, 44568, 50197,
+#'                           39113, 41668, 30144, 52897, 32129, 67686, 31768, 49094)
+#' )
+#'
+#' sales
+#'
+#' # We can call `fill()` and only need to specify the column to fill since the
```

Comment:
I think overall you've provided a bit too much explanation (which would be suitable for a vignette, but not docs) — can you please make it a bit more concise?